File: 2023/09/09.py
from sys import stdin

lines = stdin.read().strip().split('\n')

first = last = 0
for line in lines:
    diff = [int(a) for a in line.split()]
    sign = 1
    while any(diff):
        last += diff[-1]
        first += diff[0] * sign
        sign *= -1
        diff = [b - a for a, b in zip(diff, diff[1:])]

print(first, last, sep='\n')

File: 2023/09/09.rs
use std::io::{stdin, Read};
use itertools::Itertools;

fn main() {
    let mut input = String::new();
    stdin().read_to_string(&mut input).unwrap();

    let mut first = 0;
    let mut last = 0;
    for line in input.lines() {
        let mut diff = line.split(' ').map(|n| n.parse().unwrap()).collect::<Vec<i32>>();
        let mut sign = 1;
        while let [a, .., z] = *diff {
            first += a * sign;
            last += z;
            sign *= -1;
            diff = diff.iter().tuple_windows().map(|(a, b)| b - a).collect();
        }
    }
    println!("{}\n{}", last, first);
}